Mattias Hedbom

Hedbom98 opened the pull request.

Commit:
domain model completed

Commit:
domain model, small adjustment

Commit:
domain model, small adjustment

Commit:
created all classes, also made som changes to the domain model

Commit:
Created the initial test for the seePrice() method in menu

Commit:
Created the initial seePrice() method

Commit:
Completed the seePrice() method/ test.

Commit:
initial method for showAllFillingsWithCosts() method

Commit:
Needed to add String typeOfItem to Item class, needed to list all fillings/ bagels etc.

Commit:
Needed to add String typeOfItem to Item class, needed to list all fillings/ bagels etc.

Also changed the return-value of showAllFillingsWithCosts()

Commit:
completed method for showAllFillingsWithCosts()

Commit:
completed method for isContainedInInventory()

Commit:
initial method for isContainedInInventory completed

Commit:
Method for isContainedInInventory completed

Commit:
Updated the fields and the constructor for the basket-class

Commit:
Initial test for addItem() in basket-class completed

Commit:
Initial method for addItem() in basket-class completed

Commit:
Method for addItem() in basket-class completed, also added an extra test for the same method

Commit:
Initial method for removeItem() in basket-class completed

Commit:
Initial method for removeItem() in basket-class completed

Commit:
Test for removeItem() in basket-class completed

Commit:
Method for removeItem() in basket-class completed

Commit:
Initial test for changeSizeOfBasket() in basket-class completed

Commit:
Initial method for changeSizeOfBasket() in basket-class completed. Also identified an error in my domain model that I corrected.

Commit:
Method/ Test for changeSizeOfBasket() in basket-class completed.

Commit:
Initial Test for totalCost() in basket-class completed.

Commit:
Initial method for totalCost() in basket-class completed.

Commit:
Method / Test for totalCost() in basket-class completed.

Commit:
Domain model adjusted for extension 1. Tests made for method totalCostWithDiscounts();

ninja-aruna commented on src/main/java/com/booleanuk/core/Menu.java:
```diff
@@ -0,0 +1,41 @@
+        String allFillingsWithPrices = "";
+        for(Item anItem : this.itemsOnMenu){
+            if(anItem.getTypeOfItem().equals("Filling")){
+                allFillingsWithPrices += anItem.getName() + ", " + anItem.getPrice() + "$\n";
```
Use stringbuilder in place of concatenating strings in a loop, as strings are immutable, so a new string will be created at each iteration

Commit:
Forgot to write the code for the extension in the extension folder, thus I created two new classes and deleted the "extension-code" from the core exercise

Commit:
Forgot the discount for 1bagel and 1 coffee, added another test for this

Commit:
Forgot the discount for 1bagel and 1 coffee, completed the method to fulfill this requirement

Commit:
Added the class diagrams, the one made before the extension and the one used for the extension (1 & 2). Also updated the domain model with the receipt class

Commit:
made a small adjustment to the receipt class

Commit:
Created the test for the second extension, also created the ExtensionReceipt class

Commit:
redid the domain model for the receipt class, added two more methods:
totalCostWithDiscounts()
costWithDiscounts()

Updated the totalCostWithDiscounts() from the first extension so it accepts a HashMap instead. Created new tests for this new method

Commit:
Test made for the costWithDiscounts() method

Commit:
method costWithDiscounts() completed

Commit:
Discovered that I needed two parameters in my costWithDiscounts() method, therefore I needed to slightly change the domain model, the test and lastly the method

Commit:
Method printReceipt() completed

Commit:
Forgot to update the classDiagram for the extension part when I recently altered the domain model
